### PR TITLE
Support data layout strategy generation in scheduler

### PR DESCRIPTION
**Pull request by teamurko**
## Summary

Add SLO type to scheduler.
This PR builds on top of https://github.com/linkedin/openhouse/pull/158

## Changes

- [ ] Client-facing API Changes
- [ ] Internal API Changes
- [ ] Bug Fixes
- [x] New Features
- [ ] Performance Improvements
- [ ] Code Style
- [ ] Refactoring
- [ ] Documentation
- [ ] Tests

For all the boxes checked, please include additional details of the changes made in this pull request.  

## Testing Done

- [x] Manually Tested on local docker setup. Please include commands ran, and their output.
- [x] Added new tests for the changes made.
- [ ] Updated existing tests to reflect the changes made.
- [ ] No tests added or updated. Please explain why. If unsure, please feel free to ask for help.
- [ ] Some other form of testing like staging or soak time in production. Please explain.

For all the boxes checked, include a detailed description of the testing done for the changes made in this pull request.
Create and populate tables
```
scala> spark.sql("create table u_openhouse.dlo_skip (id int, data string)").show()
++
||
++
++
scala> spark.sql("create table u_openhouse.dlo_run (ts timestamp, data string) partitioned by (days(ts))").show()
++
||
++
++
scala> spark.sql("insert into u_openhouse.dlo_run values (current_timestamp(), 'data')").show()
++
||
++
++
scala> spark.sql("insert into u_openhouse.dlo_run values (current_timestamp(), 'data')").show()
++
||
++
++
scala> spark.sql("insert into u_openhouse.dlo_run values (current_timestamp(), 'data')").show()
++
||
++
++
```
Run scheduler
```
$ docker compose --profile with_jobs_scheduler run openhouse-jobs-scheduler - --type DATA_LAYOUT_STRATEGY_GENERATION --cluster local --tablesURL http://openhouse-tables:8080 --jobsURL http://openhouse-jobs:8080
```

# Additional Information

- [ ] Breaking Changes
- [ ] Deprecations
- [ ] Large PR broken into smaller PRs, and PR plan linked in the description.

For all the boxes checked, include additional details of the changes made in this pull request.

### CODE REVIEW

**Review by maluchari — DISMISSED** [review]

File: apps/spark/src/main/java/com/linkedin/openhouse/jobs/client/TablesClient.java [hunk 1]
```diff
@@ -89,6 +89,20 @@ protected GetTableResponseBody getTable(TableMetadata tableMetadata) {
...
 
+  /**
+   * Checks if data layout strategies can be generated on the input table.
+   *
+   * @param tableMetadata table metadata
+   * @return true if data layout strategies can be generated on the table, false otherwise
+   */
+  public boolean canRunDataLayoutStrategyGeneration(TableMetadata tableMetadata) {
+    GetTableResponseBody response = getTable(tableMetadata);
+    if (response == null || response.getTimePartitioning() == null) {
```
Comment (autumnust) [inline apps/spark/src/main/java/com/linkedin/openhouse/jobs/client/TablesClient.java:100]: partitioning table is excluded ?

**Review by autumnust — CHANGES_REQUESTED** [review]

File: apps/spark/src/main/java/com/linkedin/openhouse/jobs/client/TablesClient.java (hunk 1 above)
Comment (teamurko) [reply apps/spark/src/main/java/com/linkedin/openhouse/jobs/client/TablesClient.java:100]: It's the opposite, although if the response is empty, we skip too, so sometimes we skip partitioned tables

Comment (sumedhsakdeo) [reply apps/spark/src/main/java/com/linkedin/openhouse/jobs/client/TablesClient.java:100]: Need to handle this as well ?https://github.com/linkedin/openhouse/blob/83b3797c7c278345457d9b43ac16ce1e26a9ceab/services/tables/src/main/java/com/linkedin/openhouse/tables/api/spec/v0/response/GetTableResponseBody.java#L91

Comment (teamurko) [reply apps/spark/src/main/java/com/linkedin/openhouse/jobs/client/TablesClient.java:100]: good catch, fixed

**Pull request merged**